cardano-node/src/Cardano/Tracing/Tracers.hs

```diff
@@ -179,10 +179,13 @@ instance ElidingTracer (WithSeverity (ChainDB.TraceEvent blk)) where
   doelide (WithSeverity _ (ChainDB.TraceCopyToImmDBEvent _)) = True
   doelide _ = False
   conteliding _tverb _tr _ (Nothing, _count) = return (Nothing, 0)
-  conteliding tverb tr ev@(WithSeverity _ (ChainDB.TraceAddBlockEvent (ChainDB.AddedToCurrentChain _ _ _))) (_old, count) = do
-      when (count > 0 && count `mod` 250 == 0) $ do  -- report every 250th block
-          traceWith (toLogObject' tverb tr) ev
-      return (Just ev, count + 1)
+  conteliding tverb tr ev@(WithSeverity _ (ChainDB.TraceAddBlockEvent (ChainDB.AddedToCurrentChain pt _ _))) (_old, oldt) = do
+      let slotno = toInteger $ unSlotNo (realPointSlot pt)
+      tnow <- fromIntegral <$> getMonotonicTimeNSec
+      let deltat = tnow - oldt
+      when (deltat > 250000000 || slotno `mod` 200 == 0) $  -- report at most every 250 ms or every 200th block
+           traceWith (toLogObject' tverb tr) ev
+      return (Just ev, tnow)
```

Comment:
I don't think that you should always return `tnow` here.
Shouldn't it be something like:
```
      if (deltat > 250000000 || slotno `mod` 200 == 0)  -- report at most every 250 ms or every 200th block
          then do
            traceWith (toLogObject' tverb tr) ev
            return (Just ev, tnow)
          else return (Just ev, oldt)
```
so that you only update the old timestamp when you actually log something.

Reply:
it's the timestamp of the last event received so measuring the time between events. You propose to measure the time between print outputs.
In chain-sync mode it suppresses messages and only shows every 200th block whereas in chain following mode it will print all blocks which arrive every 20 seconds. With your proposed change it will print four messages a second in chain-sync mode; no change in chain following mode. Can this be true?

Reply:
Imagine a very slow peer, syncing at a rate of 4 blocks per second.
Since the time between events is never less then or equal to 250ms the only time a message will be displayed is once every 50s.

Reply:
@karknu:  tested your proposed controlling of the output cadence, and it looks really nice and simpler.
```
{"at":"2020-04-30T13:54:55.04Z","env":"1.11.0:76b49","ns":["cardano.node.ChainDB"],"data":{"kind":"TraceAddBlockEvent.AddedToCurrentChain","newtip":"7079ec6@4100334"},"app":[],"msg":"","pid":"27005","loc":null,"host":"AlexBook","sev":"Notice","thread":"33"}
{"at":"2020-04-30T13:54:56.39Z","env":"1.11.0:76b49","ns":["cardano.node.ChainDB"],"data":{"kind":"TraceAddBlockEvent.AddedToCurrentChain","newtip":"b40f72d@4101235"},"app":[],"msg":"","pid":"27005","loc":null,"host":"AlexBook","sev":"Notice","thread":"33"}
{"at":"2020-04-30T13:54:57.64Z","env":"1.11.0:76b49","ns":["cardano.node.ChainDB"],"data":{"kind":"TraceAddBlockEvent.AddedToCurrentChain","newtip":"21e4b6d@4102197"},"app":[],"msg":"","pid":"27005","loc":null,"host":"AlexBook","sev":"Notice","thread":"33"}
{"at":"2020-04-30T13:55:15.53Z","env":"1.11.0:76b49","ns":["cardano.node.ChainDB"],"data":{"kind":"TraceAddBlockEvent.AddedToCurrentChain","newtip":"0e1ccae@4102591"},"app":[],"msg":"","pid":"27005","loc":null,"host":"AlexBook","sev":"Notice","thread":"33"}
{"at":"2020-04-30T13:55:32.68Z","env":"1.11.0:76b49","ns":["cardano.node.ChainDB"],"data":{"kind":"TraceAddBlockEvent.AddedToCurrentChain","newtip":"9e3bdc5@4102592"},"app":[],"msg":"","pid":"27005","loc":null,"host":"AlexBook","sev":"Notice","thread":"33"}
{"at":"2020-04-30T13:55:54.57Z","env":"1.11.0:76b49","ns":["cardano.node.ChainDB"],"data":{"kind":"TraceAddBlockEvent.AddedToCurrentChain","newtip":"f2a78d1@4102593"},"app":[],"msg":"","pid":"27005","loc":null,"host":"AlexBook","sev":"Notice","thread":"33"}
{"at":"2020-04-30T13:56:14.07Z","env":"1.11.0:76b49","ns":["cardano.node.ChainDB"],"data":{"kind":"TraceAddBlockEvent.AddedToCurrentChain","newtip":"4fecdac@4102594"},"app":[],"msg":"","pid":"27005","loc":null,"host":"AlexBook","sev":"Notice","thread":"33"}
{"at":"2020-04-30T13:56:34.65Z","env":"1.11.0:76b49","ns":["cardano.node.ChainDB"],"data":{"kind":"TraceAddBlockEvent.AddedToCurrentChain","newtip":"b882295@4102595"},"app":[],"msg":"","pid":"27005","loc":null,"host":"AlexBook","sev":"Notice","thread":"33"}
{"at":"2020-04-30T13:56:52.08Z","env":"1.11.0:76b49","ns":["cardano.node.ChainDB"],"data":{"kind":"TraceAddBlockEvent.AddedToCurrentChain","newtip":"db0e9bc@4102596"},"app":[],"msg":"","pid":"27005","loc":null,"host":"AlexBook","sev":"Notice","thread":"33"}
```
after full chain-sync it goes into chain following mode. the log files is just 1 MB in size.
so this mode works great too and is independent of the sync rate.